### ISSUE TITLE
false-positive type casting is not necessary

### ISSUE THREAD

**Issue by m8rge** [kalessil/phpinspectionsea#992]
Let's start with common details:
 
| Subject    | Details                                                  |
| :--------- | :------------------------------------------------------- |
| Issue type | Bug/False-Positive     |
| IDE        | PhpStorm Build #PS-182.3684.40                          |
| Plugin     | Php Inspections EA Extended 3.0.5.1 |

Description or screenshot:
--
![image](https://user-images.githubusercontent.com/575777/43508548-acac5d4a-9589-11e8-8940-3b69c2f72f0f.png)
`$this->data` is associative array with following declaration in parent class:
```
    /**
     * @var array
     */
    protected $data = [];
```
So, `$this->data['order_id']`, can be mixed type and type casting is required.

Full code in https://github.com/kalessil/phpinspectionsea/issues/992#issuecomment-411150305

Environment details
---
PhpStorm 2018.2 EAP
Build #PS-182.3684.40, built on July 17, 2018
PhpStorm EAP User
Expiration date: August 16, 2018
JRE: 1.8.0_152-release-1248-b8 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6

**Comment by ktomk** [comment]
Even this is a false positive, the int cast belongs to `$this->data['order_id']` only,

So please also additionally report if casting only for that variable will make any difference.

**Comment by m8rge** [comment]
Hi @ktomk, In my case int cast refer to whole expression: `$this->data['order_id'] ?? 0`

**Comment by ktomk** [comment]
@m8rge sure, but part of it (the 0) is already an integer and must not be casted. That is perhaps triggering it so changing the arrangement might have turned out more however if I understand this case right, it is not possible to change the arrangement as `$this->data['order_id']` might be unset and would trigger a warning.

**Comment by funivan** [comment]
Hello @m8rge 
I`m using  Php plugin version: 182.3911.19 and Plugin PhpInspections EA Version: 2.0.7
Can not reproduce this issue. 
Maybe someone rewrites the type of `$this->data `

Can you check this code:
```php
class A
{
    /** @var array */
    private $data = [];
    public function show(): void {
        $name = (int)($this->data['order_id'] ?? 0);
        echo $name;
    }
}
```

**Comment by m8rge** [comment]
Hi @funivan,

Your example works correct. I narrow down issue to this:
```php
class B extends A
{
    public function show(): void
    {
        $name = (int)($this->data['order_id'] ?? 0);
        echo $name;
    }
}

class A
{
    /** @var array */
    protected $data = [];
}
```

Class extending is required to reproduce bug.

**Comment by funivan** [comment]
Confirm

**Comment by kalessil** [comment]
Fixed!
